### PR TITLE
chunked: add fuzzing tests for the cache

### PR DESCRIPTION
**Pull request by giuseppe**
and fix some issues found with the new tests.

### CODE REVIEW

**Comment by cgwalters** [comment]
I was unaware of this whole subsystem until now.  This is the thing keeping indicies of chunks that are used for inside-a-file partial pulls?

The changes seem sane superficially.

**Comment by giuseppe** [comment]
> I was unaware of this whole subsystem until now. This is the thing keeping indicies of chunks that are used for inside-a-file partial pulls?

yes, this is needed to avoid parsing the json metadata files each time we lookup for a digest.  In the same cache file we store the files payload digest, the chunks digest and the hard link fingerprint.  The hard link fingerprint a digest that is used to find files that can be used for hard links deduplication (payload digest + UID + GID + mode + xattrs).

File: pkg/chunked/cache_linux.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ const (
 	// https://pages.cs.wisc.edu/~cao/papers/summary-cache/node8.html
 	bloomFilterScale  = 10 // how much bigger is the bloom filter than the number of entries
 	bloomFilterHashes = 3  // number of hash functions for the bloom filter
+
+	maxTagsLen = 100 * 1024 * 1024 // 100MB max size for tags len
```
Comment (rhatdan) [inline pkg/chunked/cache_linux.go:38]: Why not build a Const for this magic number since the two PRs both create it.

Comment (giuseppe) [reply pkg/chunked/cache_linux.go]: I've changed it to use the go-units package.

**Pull request merged**